timetypes/duration_value.go

```diff
@@ -0,0 +1,196 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package timetypes
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/attr/xattr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/function"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+)
+
+var (
+	_ basetypes.StringValuable       = (*Duration)(nil)
+	_ xattr.ValidateableAttribute    = (*Duration)(nil)
+	_ function.ValidateableParameter = (*Duration)(nil)
+)
+
+// Duration represents a valid Go time duration string.
+// See https://pkg.go.dev/time#ParseDuration for more details
+type Duration struct {
```

Comment:
Similarly with the value type

```suggestion
type GoDuration struct {
```

Along with changes elsewhere.

Reply:
Done :heavy_check_mark: